src/utiltime.cpp

```diff
@@ -77,13 +77,13 @@ void MilliSleep(int64_t n)
 
 std::string DateTimeStrFormat(const char* pszFormat, int64_t nTime)
 {
-    static std::locale classic(std::locale::classic());
-    // std::locale takes ownership of the pointer
-    std::locale loc(classic, new boost::posix_time::time_facet(pszFormat));
-    std::stringstream ss;
-    ss.imbue(loc);
-    ss << boost::posix_time::from_time_t(nTime);
-    return ss.str();
+    char buf[256];
+    struct tm time_struct;
+    time_t time_val = nTime;
+    gmtime_r(&time_val, &time_struct);
```

Comment:
Is this portable? Afiak all of the madness is usually an abstraction of these variants.

Reply:
Well at least this code doesn't need any allocations.

Reply:
Why don't just use `gmtime`? It's more portable.

Reply:
@ken2812221 `gmtime` is not thread safe. We either user `gmtime_r`, or introduce a lock around the call.